pandas/core/series.py

```diff
@@ -2503,26 +2501,34 @@ def to_period(self, freq=None, copy=True):
                                  index=new_index).__finalize__(self)
 
     #------------------------------------------------------------------------------
+    # string methods
+
+    def _make_str_accessor(self):
+        return StringMethods(self)
```

Comment:
I had the same question. I use ``core.common.is_object_dtype`` (of course nice error message and tests)